src/parse-result.ts

```diff
@@ -982,6 +982,12 @@ export default class ParseResult {
               quote = '"';
             } else if (wasQuotableValue && !newValueNeedsQuotes) {
               quote = '';
+            } else if (
+              ast.value.type === 'ConcatStatement' &&
+              isSyntheticWithNoLocation(ast.value)
```

Comment:
```suggestion
              ast.value.type === 'ConcatStatement' &&
              ast.value.loc.module === '(synthetic)'
```
I think we may want to match every synthetic value (even if it was provided a loc when created).

Reply:
```suggestion
              !newValueNeedsQuotes &&
              ast.value.loc.module === '(synthetic)'
```
!newValueNeedsQuotes is an alias for `ast.value.type === 'ConcatStatement'`

Reply:
(aprat from that small detail, LGTM)